### DISCUSSION TITLE
How to get a list of rpc urls from a wagmiClient for a given chainId?

### DISCUSSION THREAD

**Discussion by ryanberckmans** [wevm/wagmi#2179] [Q&A] [answered]
If I construct `const wagmiClient = createClient(...);`, and then later, I have a `const chainId = 5`, how can I retrieve an array of prioritized rpc urls for that chainId?

Eg.

```typescript
const rpcUrls: string[] = wagmiClient.???(chainId);
// rpcUrls = ["... alchemy", "...infura", "...public rpc from Chain"];
```

**Comment by SaadNiaziDev** [comment, marked as answer]
``
const provider = useProvider( {
		chainId: xyz
	} )
``
then console log this provider u will see all details in provider.chains

**Reply by ryanberckmans** [reply]
thanks, I see this correctly produces the final chains object that has the final rpcUrls.
